File: windows/src/SkillHub.Core/ToolSource.cs
namespace SkillHub.Core;

public enum ToolSource
{
    CursorUser,
    CursorBuiltin,
    Claude,
    Codex,
    Agents,
    Proma,
    PromptLibrary,
    CodexPrompts,
    Custom
}

public static class ToolSourceInfo
{
    public static readonly ToolSource[] SkillRoots =
    [
        ToolSource.CursorUser,
        ToolSource.CursorBuiltin,
        ToolSource.Claude,
        ToolSource.Codex,
        ToolSource.Agents,
        ToolSource.Proma
    ];

    public static string Title(this ToolSource source) => source switch
    {
        ToolSource.CursorUser => "Cursor",
        ToolSource.CursorBuiltin => "Cursor 内置",
        ToolSource.Claude => "Claude",
        ToolSource.Codex => "Codex",
        ToolSource.Agents => "Agents",
        ToolSource.Proma => "Proma",
        ToolSource.PromptLibrary => "Skill Hub 库",
        ToolSource.CodexPrompts => "Codex Prompts",
        ToolSource.Custom => "自定义",
        _ => source.ToString()
    };

    public static string Glyph(this ToolSource source) => source switch
    {
        ToolSource.CursorUser => "◎",
        ToolSource.CursorBuiltin => "◌",
        ToolSource.Claude => "✶",
        ToolSource.Codex => "</>",
        ToolSource.Agents => "👥",
        ToolSource.Proma => "✦",
        ToolSource.PromptLibrary => "▤",
        ToolSource.CodexPrompts => "❝",
        _ => "📁"
    };

    public static bool IsWritable(this ToolSource source) => source != ToolSource.CursorBuiltin;

    public static bool IsSkillRoot(this ToolSource source) => source switch
    {
        ToolSource.CursorUser or ToolSource.CursorBuiltin or ToolSource.Claude
            or ToolSource.Codex or ToolSource.Agents or ToolSource.Proma => true,
        _ => false
    };

    public static string TintHex(this ToolSource source) => source switch
    {
        ToolSource.CursorUser => "#0A84FF",
        ToolSource.CursorBuiltin => "#8E8E93",
        ToolSource.Claude => "#FF9F0A",
        ToolSource.Codex => "#30D158",
        ToolSource.Agents => "#5E5CE6",
        ToolSource.Proma => "#FF375F",
        ToolSource.PromptLibrary => "#A2845E",
        ToolSource.CodexPrompts => "#64D2FF",
        _ => "#8E8E93"
    };
}

File: SkillHub/Catalog/Models/ToolSource.swift
import Foundation

/// Where a skill or prompt lives on disk, expressed as the tool that owns that folder.
nonisolated enum ToolSource: String, Codable, CaseIterable, Sendable, Hashable {
    case cursorUser
    case cursorBuiltin
    case claude
    case codex
    case agents
    case proma
    case promptLibrary
    case codexPrompts
    case custom

    var title: String {
        switch self {
        case .cursorUser: "Cursor"
        case .cursorBuiltin: "Cursor 内置"
        case .claude: "Claude"
        case .codex: "Codex"
        case .agents: "Agents"
        case .proma: "Proma"
        case .promptLibrary: "Skill Hub 库"
        case .codexPrompts: "Codex Prompts"
        case .custom: "自定义"
        }
    }

    var symbolName: String {
        switch self {
        case .cursorUser: "cursorarrow.rays"
        case .cursorBuiltin: "cursorarrow.square"
        case .claude: "sparkles"
        case .codex: "chevron.left.forwardslash.chevron.right"
        case .agents: "person.2"
        case .proma: "wand.and.stars"
        case .promptLibrary: "books.vertical"
        case .codexPrompts: "text.quote"
        case .custom: "folder"
        }
    }

    /// Cursor's bundled skills are managed by Cursor itself and must never be edited or deleted.
    var isWritable: Bool { self != .cursorBuiltin }

    var isSkillRoot: Bool {
        switch self {
        case .cursorUser, .cursorBuiltin, .claude, .codex, .agents, .proma: true
        case .promptLibrary, .codexPrompts, .custom: false
        }
    }

    static let skillRoots: [ToolSource] = [.cursorUser, .cursorBuiltin, .claude, .codex, .agents, .proma]
}
